File: app/src/main/java/han/ayeon/searchimgandvideo/view/MainActivity.java
package han.ayeon.searchimgandvideo.view;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SearchView;
import androidx.appcompat.widget.Toolbar;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.tabs.TabLayout;

import han.ayeon.searchimgandvideo.R;
import han.ayeon.searchimgandvideo.view.savedlist.SavedListFragment;
import han.ayeon.searchimgandvideo.view.searchlist.SearchListFragment;


public class MainActivity extends AppCompatActivity {

    private SearchTabPagerAdapter pagerAdapter;

    private SearchListFragment searchListFragment;
    private SavedListFragment savedListFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TabLayout tabLayout = findViewById(R.id.tabLayout);
        ViewPager viewPager = findViewById(R.id.viewPager);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        searchListFragment = SearchListFragment.newInstance();
        savedListFragment = SavedListFragment.newInstance();

        pagerAdapter = new SearchTabPagerAdapter(getSupportFragmentManager());
        setupViewPager(viewPager);
        tabLayout.setupWithViewPager(viewPager);
        viewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(tabLayout));

    }


    private void setupViewPager(ViewPager viewPager) {
        pagerAdapter.addFragment(searchListFragment, "검색 결과");
        pagerAdapter.addFragment(savedListFragment, "내 보관함");
        viewPager.setAdapter(pagerAdapter);
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.search, menu);
        MenuItem searchMenu = menu.findItem(R.id.menu_search);
        final SearchView searchView = (SearchView) searchMenu.getActionView();
        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                searchListFragment.search(query);
                searchView.clearFocus();
                return false;
            }
            @Override
            public boolean onQueryTextChange(String newText) {
                return false;
            }
        }

        );

        return super.onCreateOptionsMenu(menu);
    }

}

File: app/src/main/java/han/ayeon/searchimgandvideo/view/MainActivity.kt
package han.ayeon.searchimgandvideo.view

import android.os.Bundle
import android.view.Menu
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.SearchView
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import han.ayeon.searchimgandvideo.R
import han.ayeon.searchimgandvideo.view.savedlist.SavedListFragment
import han.ayeon.searchimgandvideo.view.searchlist.SearchListFragment
import kotlinx.android.synthetic.main.activity_main.*

/**
 * Created by HanAYeon on 2019-07-18.
 */

class MainActivity : AppCompatActivity() {

    private lateinit var pagerAdapter: SearchTabPagerAdapter
    private lateinit var searchListFragment : SearchListFragment
    private lateinit var savedListFragment : SavedListFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main)

        setSupportActionBar(toolbar)

        searchListFragment = SearchListFragment().newInstance()
        savedListFragment = SavedListFragment().newInstance()

        pagerAdapter = SearchTabPagerAdapter(supportFragmentManager)



        setUpViewPager(viewPager)
        tabLayout.setupWithViewPager(viewPager)
        viewPager.addOnPageChangeListener(TabLayout.TabLayoutOnPageChangeListener(tabLayout))

    }


    private fun setUpViewPager(viewPager : ViewPager) {
        pagerAdapter.addFragment(searchListFragment, "검색 결과")
        pagerAdapter.addFragment(savedListFragment, "내 보관함")
        viewPager.adapter = pagerAdapter
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.search, menu)
        val searchMenu = menu!!.findItem(R.id.menu_search)
        val searchView = searchMenu.actionView as SearchView
        searchView.setOnQueryTextListener(object: SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String?): Boolean {
                searchListFragment.search(query!!)
                searchView.clearFocus()
                return false
            }

            override fun onQueryTextChange(newText: String?): Boolean {
                return false
            }

        })

        return super.onCreateOptionsMenu(menu)
    }
}